text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,502 @@
+
+**CloudFormationStackSetArtifact**
+* **[Type](https://github.com/aws/aws-cdk/blob/5a6fa7fa17a5dce5e429eed4ebfe2dbbac3d6d07/packages/%40aws-cdk/cloud-assembly-schema/lib/cloud-assembly/schema.ts#L17)**: “aws:cloudformation:stack-set”
+    * A new artifact type classification
+* **Properties**:
+  * **templateFile**: File path to generated template
+  * **description**: Description of stack set
+  * **administrationRoleName**: Role that assumes the executionRole in the child account
+  * **executionRoleName**: Role in child account that performs the changes
+  * **permissionModel**: For this MVP, only self-managed will be supported
+  * **operationPreferences**:
+    * **faultToleranceCount/faultTolerancePercentage:** (1-n or 1-100)
+      * Only one should be supplied
+    * **maxConcurrentCount/maxConcurrentPercentage:** (1-n or 1-100)
+      * Only one should be supplied
+    * **regionConcurrencyType:** SEQUENTIAL/PARALLEL
+
+References:
+* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/cx-api/lib/artifacts/cloudformation-artifact.ts
+* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/cloud-assembly-schema/lib/cloud-assembly/artifact-schema.ts
+* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/cloud-assembly-schema/schema/cloud-assembly.schema.json
+
+### [2.2] CDK Stack Set Construct
+Users should be able to to define infrastructure resources in the same way today. AWS infrastructure constructs and synthesis logic are tightly coupled with the “core.Stack” type, where they must be created the the scope of “core.Stack” or utilize “core.Stack” types in logic. Adding a new Stack Set core construct that generates templates in that same way as Stack may require a large amount of refactoring.
+
+**Option 1:** Create a new core.StackSet construct that extends Stack or Abstract Class that StackSet/Stack inherit from **(Preferred)**
+
+**Example Usage:**
+```ts
+class POCStackSet extends cdk.StackSet {
+    constructor(scope: cdk.App, id: string, props: cdk.StackSetProps = {}) {
+        super(scope, id, props);
+
+        new Role(this, 'ExampleRole', {
+          assumedBy: new ServicePrincipal('s3.amazonaws.com')
+        });
+    }
+}
+new POCStackSet(app, 'poc-stack-set',
+    {
+        faultTolerancePercentage: 10,
+        maxConcurrentPercentage: 10,
+        regionConcurrencyType: 'PARALLEL'
+        .
+        .
+        .
+     }
+);
+```
+**Synthesis Flow:**
+
+![Architecture Diagram](../images/self-managed-stack-sets/cdk_stack_set_construct_synth_flow_option_1.jpg)
+* User extends the core StackSet construct, and defines infrastructure resources
+* A new synthesizer adds the new stack set artifact type to the manifest
```

Comment:
Not sure why do we need a new synthesizer? Why not modify the existing one(s) to handle this?

Reply:
So writing a POC, I ended up extending `DefaultStackSynthesizer` as `BootstraplessStackSetSynthesizer`

Reply:
Talked offline, required for emitting additional artifact type and properties

Reply:
If you could put in the RFC that justification why is a new synthesizer needed, that would be great 🙂.